__tests__/app.test.js

```diff
@@ -48,4 +48,49 @@ describe("GET /api/topics", () => {
         })
       })
   });
+});
+
+describe("GET /api/articles/:article_id", () => {
+  test("200: Responds with a status code", () => {
+    return request(app)
+      .get("/api/articles/1")
+      .expect(200)
+  });
+
+  test("404: Responds with Article Not Found when the article ID does not exist", () => {
+    return request(app)
+      .get("/api/articles/9999")
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe('Article not found')
+      })
+  });
+
+  test("400: Responds with Invalid Article ID when the article ID is not a number", () => {
+    return request(app)
+      .get("/api/articles/abcdefg")
+      .expect(400)
+      .then(({body}) => {
+        expect(body.msg).toBe('Invalid article ID')
+      });
+  });
+
+  test("200: Responds with an article object, which should have all the properties.", () => {
+    return request(app)
+      .get("/api/articles/2")
+      .expect(200)
+      .then((response) => {
+        const {body} = response
+        const {article} = body
+          expect(article).toMatchObject({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
```

Comment:
Do we want the article_id to be any number? if the number is 5, this test would still pass, even though it shouldn't! how could we make this more robust?

Reply:
Otherwise a good test for flexibility - I'm not really concerned about the values of the other keys personally! the test data could always be updated but this request should always return article 2